frontend/components/customizer/query-string-builder.js

```diff
@@ -36,255 +36,246 @@ function getBadgeOption(name) {
   return supportedBadgeOptions.find(opt => opt.name === name)
 }
 
+function getQueryString({ queryParams, badgeOptions }) {
+  const outQuery = {}
+  let isComplete = true
+
+  Object.entries(queryParams).forEach(([name, value]) => {
+    // As above, there are two types of supported params: strings and
+    // booleans.
+    const isStringParam = typeof value === 'string'
+    if (isStringParam) {
+      if (value) {
+        outQuery[name] = value
+      } else {
+        // Skip empty params.
+        isComplete = false
+      }
+    } else {
+      // Generate empty query params for boolean parameters by translating
+      // `{ compact_message: true }` to `?compact_message`. When values are
+      // false, skip the param.
+      if (value) {
+        outQuery[name] = null
+      }
+    }
+  })
+
+  Object.entries(badgeOptions).forEach(([name, value]) => {
+    const { shieldsDefaultValue } = getBadgeOption(name)
+    if (value && value !== shieldsDefaultValue) {
+      outQuery[name] = value
+    }
+  })
+
+  const queryString = stringifyQueryString(outQuery)
+
+  return { queryString, isComplete }
+}
+
+function ServiceQueryParam({
+  name,
+  value,
+  exampleValue,
+  isStringParam,
+  stringParamCount,
+  handleServiceQueryParamChange,
+}) {
+  return (
+    <tr>
```

Comment:
Apologies as I think I've asked this same question before in other threads 😄, but we no longer need the `key={name}` attribute with this new approach right?

Reply:
The key is used to disambiguate components at the same level. Since this returns only one component, it doesn't have any effect. The `<ServiceQueryParam>` invocation sets a key instead, as that's where the array of components is being rendered.